### PR TITLE
chore(core): pnpm 

### PR DESCRIPTION
**Pull request by etowahadams**
Fix #
Toward #897

## Change List
 - Replaces `yarn` with `pnpm`  as the main package manager. 

We use pnpm workspaces in Guisling and other projects in the lab for monorepo package management. To be consistent, Gosling should also use pnpm. This is in preparation for making Gosling a monorepo.  

## Checklist
 - [x] Ensure the PR works with all demos on the online editor
 - [ ] Unit tests added or updated
 - [ ] Examples added or updated
 - [ ] [Documentation](https://github.com/gosling-lang/gosling-website) updated (e.g., added API functions)
 - [ ] Screenshots for visual changes (e.g., new encoding support or UI change on Editor)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,6 +65,7 @@
         "d3-array": "^2.5.1",
         "d3-color": "^2.0.0",
         "d3-dsv": "^2.0.0",
+        "d3-format": "^3.1.0",
```
Comment (etowahadams) [inline package.json:69]: Because pnpm doesn't hoist package dependencies in `node_modules`, I needed to add several packages.

File: package.json [hunk 2]
```diff
@@ -132,6 +141,7 @@
         "react-resize-detector": "^4.2.3",
         "react-router-dom": "^5.2.0",
         "remark-gfm": "^1.0.0",
+        "safe-stable-stringify": "^2.4.3",
```
Comment (etowahadams) [inline package.json:145]: Used when running `pnpm schema`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -5,15 +5,15 @@ While contribution guidelines are loosely defined, we recommend to read the foll
...
 
-After installing [Yarn](https://yarnpkg.com/getting-started/install), clone this repository and run the following commands to install all dependencies and run the Gosling.js editor locally. We recommend you use Node 20. 
+After installing [pnpm](https://pnpm.io/installation), clone this repository and run the following commands to install all dependencies and run the Gosling.js editor locally. We recommend you use Node 20. 
 
 ```sh
 git clone https://github.com/gosling-lang/gosling.js.git # Clone the repository to your current directory
 cd gosling.js                                            # Navigate to gosling repository
-yarn                                                     # Install dependencies 
-yarn start                                               # Start a local server running the Gosling online editor 
+pnpm                                                     # Install dependencies
```
Comment (sehilyi) [inline CONTRIBUTING.md:15]: It looks like `pnpm` itself does not install packages?

```suggestion
pnpm install                                             # Install dependencies 
```

**Review by sehilyi — APPROVED** [review]
Awesome! Tested locally and works well. I will merge this into the master.

**Comment by etowahadams** [comment]
Great thanks!

File: CONTRIBUTING.md (hunk 3 above)
Comment (dvdkouril) [reply CONTRIBUTING.md:15]: I just noticed this, too. Seems like it slipped through the cracks and wasn't fixed in the PR?

Comment (etowahadams) [reply CONTRIBUTING.md:15]: Good catch, will make the PR now!

**Pull request merged**